integration_tests/security-providers/src/test/java/org/robolectric/integrationtests/securityproviders/SecurityProvidersTest.java

```diff
@@ -15,24 +12,14 @@
 @RunWith(RobolectricTestRunner.class)
 @Config(sdk = Config.ALL_SDKS)
 public class SecurityProvidersTest {
-  // Conscrypt (which calls into native libraries) can only be loaded once as per JNI spec:
-  // https://docs.oracle.com/javase/1.5.0/docs/guide/jni/spec/invocation.html#library_version
-  // If we try to load it into multiple class loaders, UnsatisfiedLinkErrors will be thrown.
-  private static final Provider CONSCRYPT_PROVIDER = new OpenSSLProvider();
```

Comment:
This is actually the biggest problem that I can see with using Conscrypt. Unlike BC, creating Conscrypt OpenSSLProviders may not be idempotent. If Robolectric calls `new OpenSSLProvider()`, and there's an existing test that also calls `new OpenSSLProvider()`, there will be an UnsatisfiedLinkErrors(at least on some JVMs). I haven't quite figured out a solution for this yet.

Reply:
How about consider as
1. Just leave as limitation
2. Rename package of Conscrypt such as android platform (which has `com.android` prefix) or gms security provider (which has `com.google.android.gms` prefix)

Reply:
I've reached out to the Conscrypt team about this. I think it would be a small code change for them to avoid trying to load the same native libraries more than once.